### PR TITLE
Screenshot store with MediaStore API and cleanup

### LINKED ISSUE #106: [Android 10] Replace java.io.file API
Since Android 10 `java.io.file` API is deprecated. This results in screenshots not being stored on the phone.

According to Androids [storage options summary table](https://developer.android.com/training/data-storage#java) we should pick the [MediaStore API](https://developer.android.com/reference/android/provider/MediaStore). This has two benefits:
1. Files are not being removed when the app is uninstalled
2. Other apps can access the screenshots to edit/share them

### PR DESCRIPTION
**Pull request by jrtberlin**
This fixes #106 but I don't have a device running API <= 9 with a functioning BLE stack. So this needs testing if it breaks storing images on Android 9 and below.

### CODE REVIEW

**Comment by MagneFire** [comment]
Nice! So this was the reason screenshots didn't work on Android 10 :)

Anyway I have tested this on an old phone and screenshots seem to work fine.

However, your last commit seems to break the screenshot notification. The screenshot notification does not appear on Android 9 and Android 10.

The notification did appear before that, but it was just spamming(lots of notification sounds).

**Comment by jrtberlin** [comment]
> Nice! So this was the reason screenshots didn't work on Android 10 :)
> 
> Anyway I have tested this on an old phone and screenshots seem to work fine.
> 
> ~However, your last commit seems to break the screenshot notification. The screenshot notification does not appear on Android 9 and Android 10.~
> 
> ~The notification did appear before that, but it was just spamming(lots of notification sounds).~
> 
> EDIT: It's now a silent notification! Thats what I missed. Sorry about this. Everything works perfectly

Thanks for testing. So everything works as expected on Android 10 and below?
(I switched the type to priority min to remove the notification sounds. maybe I should change that to something that's more obvious to notice, but without sound)

**Comment by MagneFire** [comment]
I haven't noticed anything weird.
- When I tap the notification it opens properly in an app.
- The progress bar moves.
- Images are properly stored under  Pictures/AsteroidOSSync/

From the user experience there appears no difference between Android 10 and below. It behaves the same.

I guess the way it is right now is fine. Maybe change the notification from silent to normal when it is downloaded. That way it looks similar to the stock Android screenshot notification.

**Comment by jrtberlin** [comment]
>  Maybe change the notification from silent to normal when it is downloaded.

This would require a second notification channel because channels [can not be edited after creation](https://developer.android.com/training/notify-user/channels).
"_After you create a notification channel, you cannot change the notification behaviors—the user has complete control at that point. Though you can still change a channel's name and description_."

I'm unsure if that's worth the bloat and additional translatable strings.

**Comment by MagneFire** [comment]
Oh, I had hoped that it would be a simple change... Guess not :disappointed: 

I guess the current way will suffice then. Adding bloat to make this work wouldn't be worth it.

File: app/src/main/java/org/asteroidos/sync/ble/ScreenshotService.java [hunk 1]
```diff
@@ -185,6 +188,63 @@ public void onEvent(ReadWriteEvent e) {
...
+                DoMediaScan(fileName);
+            } catch(IOException e) {
+                e.printStackTrace();
+            }
+            uri = FileProvider.getUriForFile(mCtx, mCtx.getApplicationContext().getPackageName() + ".fileprovider", fileName);
+        }
+        return uri;
+    }
+
+    private void DoMediaScan(File file){
```
Comment (FlorentRevest) [inline app/src/main/java/org/asteroidos/sync/ble/ScreenshotService.java:237]: nit: Isn't the convention to start method name with a lowercase letter?

File: app/src/main/java/org/asteroidos/sync/ble/ScreenshotService.java [hunk 2]
```diff
@@ -185,6 +188,63 @@ public void onEvent(ReadWriteEvent e) {
             Log.e("ScreenshotService", e.status().toString());
     }
 
+    private Uri createFile(byte[] totalData) throws IOException {
+        String dirStr = Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_PICTURES) + "/AsteroidOSSync";
+        Uri uri;
+
+        String timeStamp = new SimpleDateFormat("yyyyMMdd_HHmmss", Locale.US).format(System.currentTimeMillis());
+        File fileName;
+        fileName= new File(dirStr + "/Screenshot_" + timeStamp + ".jpg");
```
Comment (FlorentRevest) [inline app/src/main/java/org/asteroidos/sync/ble/ScreenshotService.java:197]: nit: Could you add a space before that = ? And maybe bring it on the same line as the previous one?

File: app/src/main/java/org/asteroidos/sync/ble/ScreenshotService.java [hunk 3]
```diff
@@ -185,6 +188,63 @@ public void onEvent(ReadWriteEvent e) {
...
+            metaInfo.put(MediaStore.Images.Media.DATE_TAKEN, System.currentTimeMillis());
+            Uri imageUri = resolver.insert(MediaStore.Images.Media.EXTERNAL_CONTENT_URI , metaInfo);
+            assert imageUri != null;
+            OutputStream out = resolver.openOutputStream(imageUri);
+            assert out != null;
+            try {
+                out.write(totalData);
+            } catch (IOException ioe) {
+                ioe.printStackTrace();
+            }finally {
```
Comment (FlorentRevest) [inline app/src/main/java/org/asteroidos/sync/ble/ScreenshotService.java:215]: nit: also a space before that finally for consistency

File: app/src/main/java/org/asteroidos/sync/ble/ScreenshotService.java [hunk 4]
```diff
@@ -185,6 +188,63 @@ public void onEvent(ReadWriteEvent e) {
...
+                out.close();
+                uri = imageUri;
+            }
+        }else {
```
Comment (FlorentRevest) [inline app/src/main/java/org/asteroidos/sync/ble/ScreenshotService.java:219]: nit: and a space here uh :p

**Pull request merged**